server/json_rpc.go

```diff
@@ -151,7 +151,7 @@ func StartJSONRPC(
 			if err := httpSrv.Shutdown(context.Background()); err != nil {
 				logger.Error("failed to shutdown JSON-RPC server", "error", err.Error())
 			}
-			return ln.Close()
+			return err
```

Comment:
since ln already close by [shutdown](https://github.com/golang/go/blob/release-branch.go1.23/src/net/http/server.go#L3127), but we shouldn't scope the err anyway

Reply:
I've changed that